packages/babel-generator/src/generators/flow.js

```diff
@@ -252,6 +252,16 @@ export function ExistsTypeAnnotation() {
 export function FunctionTypeAnnotation(node: Object, parent: Object) {
   this.print(node.typeParameters, node);
   this.token("(");
+
+  if (node.this) {
+    this.token("this: ");
```

Comment:
```suggestion
    this.word("this");
    this.token(":");
```
They should be printed separately because comments are allowed between `this` and `:` (They are two different tokens)
```js
function one (this /* comment */: number) {}
```